### PR TITLE
⭐️ GCP GKE cluster resource

### PR DESCRIPTION
**Pull request by imilchev**
Add GCP cluster resource. Can test it by running:
```
gcp.project.clusters{ * }
```

### CODE REVIEW

File: resources/packs/gcp/gcp.lr [hunk 1]
```diff
@@ -789,4 +790,69 @@ private gcp.dns.policy {
   enableInboundForwarding bool
   // Indicates if logging is enabled
   enableLogging bool
+}
+
+// GKE cluster resource
```
Comment (tas50) [inline resources/packs/gcp/gcp.lr]:
```suggestion
// GKE Cluster Resource
```

Comment (imilchev) [reply resources/packs/gcp/gcp.lr]: done

File: resources/packs/gcp/gcp.lr [hunk 2]
```diff
@@ -789,4 +790,252 @@ private gcp.dns.policy {
   enableInboundForwarding bool
   // Indicates if logging is enabled
   enableLogging bool
+}
+
+// GKE Cluster
+private gcp.cluster @defaults("name") {
```
Comment (chris-rock) [inline resources/packs/gcp/gcp.lr]: Cluster should be scoped in `gcp.project.cluster`. All the other resources need to be reßscoped too. I think we should also allow users to test individual clusters directly:

```
gcp.project.cluster(projectID: "myproject", name: "clustername")
```

Therefore this resource should not be private. To achieve that you need to implement the init function as done for services https://github.com/mondoohq/cnquery/blob/main/resources/packs/gcp/services.go#L99-L154

**Review by chris-rock — CHANGES_REQUESTED** [review]
This looks great. The only thing we should do is to move the resources into the `gcp.project` scope.

File: resources/packs/gcp/gcp.lr (hunk 2 above)
Comment (imilchev) [reply resources/packs/gcp/gcp.lr]: the rescoping is done. As agreed the init functions will be added at a later point. We will first add all the necessary GCP resources

**Review by chris-rock — APPROVED** [review]
Great addition @imilchev

**Pull request merged**